buildSrc/src/main/java/org/elasticsearch/gradle/Jdk.java

```diff
@@ -111,6 +111,15 @@ public TaskDependency getBuildDependencies() {
         return configuration.getBuildDependencies();
     }
 
+    public Object getBinJavaPath() {
+        return new Object() {
```

Comment:
Just a random though - not necessarily in this PR, but would it be worth defining a class like `LazyString`, so that we can clearly document why we need this pattern?

```java
import java.util.function.Supplier;

/**
 * Some important documentation goes here...
 */
public class LazyString {
    private final Supplier<String> supplier;

    public LazyString(Supplier<String> supplier) {
        this.supplier = supplier;
    }

    @Override
    public String toString() {
         return this.supplier.get();
    }
}

// Usage: new LazyString(() -> getPath() + "/bin/java");
```

Reply:
👍 absolutely

I think we did go down this route at one point. The funkiness here is that we are relying on implicit behavior that Gradle calls `toString()` on whatever we pass it. We could still do that, and perhaps implement `CharSequence` in a lazy way.